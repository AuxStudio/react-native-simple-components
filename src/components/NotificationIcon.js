import React from "react";
import { View, Text, StyleSheet } from "react-native";

import styleConstants from "../assets/styleConstants";

import Touchable from "./Touchable";

const styles = StyleSheet.create({
    notificationsButton: {
        padding: 16,
        paddingRight: 8,
    },
    notificationsContainer: {
        ...styleConstants.smallShadow,
        width: 20,
        height: 20,
        borderRadius: 10,
        overflow: "hidden", // ios
        backgroundColor: styleConstants.veryLightGrey,
        justifyContent: "center",
        alignItems: "center",
    },
    smallText: {
        fontSize: styleConstants.verySmallFont,
        color: styleConstants.white,
        marginBottom: 2,
    },
});

export default (NotificationIcon = props => {
    /*
        PROPTYPES
            handlePress     PropTypes.func
            count           PropTypes.number
    */

    const notificationsContainerStyles = props.count > 0 && {
        backgroundColor: styleConstants.secondary,
    };

    const notificationsIcon = props.handlePress ? (
        <Touchable
            onPress={props.handlePress}
            style={styles.notificationsButton}>
            <View
                style={[
                    styles.notificationsContainer,
                    notificationsContainerStyles,
                ]}>
                <Text style={[styles.smallText, styleConstants.primaryFont]}>
                    {props.count}
                </Text>
            </View>
        </Touchable>
    ) : (
        <View style={styles.notificationsButton}>
            <View
                style={[
                    styles.notificationsContainer,
                    notificationsContainerStyles,
                ]}>
                <Text style={[styles.smallText, styleConstants.primaryFont]}>
                    {props.count}
                </Text>
            </View>
        </View>
    );

    return notificationsIcon;
});
