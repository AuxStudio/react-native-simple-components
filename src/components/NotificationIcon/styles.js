import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const SIZE = 20;

const styles = StyleSheet.create({
  container: {
    width: SIZE,
    height: SIZE,
    borderRadius: SIZE / 2,
    backgroundColor: styleConstants.colors.dividerColor,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: styleConstants.fonts.sizes.verySmall,
    color: styleConstants.colors.primaryText,
  },
});

export default styles;
