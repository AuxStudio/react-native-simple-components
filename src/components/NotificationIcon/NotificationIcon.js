import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ViewPropTypes } from 'react-native';

import styles from './styles';
import styleConstants from '../../styleConstants';

const propTypes = {
  count: PropTypes.number,
  showShadow: PropTypes.bool,
  textStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {
  count: 0,
};

const CountIcon = ({ count, showShadow, textStyle, style }) => {
  const shadowStyles = showShadow && styleConstants.shadows.small;

  return (
    <View style={[styles.container, shadowStyles, style]}>
      <Text style={[styles.text, textStyle]}>{count}</Text>
    </View>
  );
};

CountIcon.propTypes = propTypes;
CountIcon.defaultProps = defaultProps;

export default CountIcon;
