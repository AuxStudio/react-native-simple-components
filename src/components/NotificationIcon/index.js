import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ViewPropTypes } from 'react-native';

import styles from './styles';
import styleConstants from '../../styleConstants';

const propTypes = {
  count: PropTypes.number,
  showShadow: PropTypes.bool,
  textStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {
  count: 0,
};

const NotificationIcon = ({ count, showShadow, textStyle, style }) => {
  const shadowStyles = showShadow && styleConstants.shadows.small;

  return (
    <View style={[styles.container, shadowStyles, style]}>
      <Text style={[styles.text, textStyle]}>{count}</Text>
    </View>
  );
};

NotificationIcon.propTypes = propTypes;
NotificationIcon.defaultProps = defaultProps;

export default NotificationIcon;
