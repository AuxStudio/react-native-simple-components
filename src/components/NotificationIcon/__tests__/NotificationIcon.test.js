import React from 'react';
import renderer from 'react-test-renderer';

import NotificationIcon from '../';

describe('NotificationIcon', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <NotificationIcon
          count={5}
          showShadow
          textStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<NotificationIcon />)).toMatchSnapshot();
  });
});
