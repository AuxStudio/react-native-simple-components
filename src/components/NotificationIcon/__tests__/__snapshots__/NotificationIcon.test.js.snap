// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationIcon renders with all props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0E0E0",
        "borderRadius": 10,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "borderColor": null,
        "borderWidth": 0,
        "elevation": 2,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.17,
        "shadowRadius": 2,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 11,
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    5
  </Text>
</View>
`;

exports[`NotificationIcon renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0E0E0",
        "borderRadius": 10,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 11,
        },
        undefined,
      ]
    }
  >
    0
  </Text>
</View>
`;
